src/Compilers/CSharp/Test/Syntax/Parsing/FuzzTesting.cs

```diff
@@ -78,5 +79,21 @@ public void c024928()
 ";
             var tree = SyntaxFactory.ParseSyntaxTree(test);
         }
+
+        [WorkItem(2771, "https://github.com/dotnet/roslyn/issues/2771")]
+        [Fact]
+        public void TestBinary()
+        {
+            // use a fixed seed so the test is reproducible
+            var random = new System.Random(12345);
+            const int n = 40 * 1000 * 1000; // 40 million "character"s
```

Comment:
Implementations of Random may use different algorithms, so it may produce different sequence on different platforms. However, I think, it would be a pretty bad Random if in 40M chars there won't be 200 bad ones. :-)